src/controllers/main.ts

```diff
@@ -184,6 +184,60 @@ export const getConnectionAndAccountController = async (
       csrfToken: csrfToken(req),
       is2faCapable,
       force2fa,
+      encrypted_password,
+    });
+  } catch (error) {
+    next(error);
+  }
+};
+
+export const getDoubleAuthenticationController = async (
```

Comment:
suggestion: This file has grown quite large. To improve maintainability and clarity, consider moving all 2FA-related logic into a separate, dedicated file.